src/protocolsupport/protocol/serializer/ItemStackSerializer.java

```diff
@@ -144,14 +161,14 @@ public static void writeItemStack(ByteBuf to, ProtocolVersion version, String lo
 			}
 			return;
 		}
-
+		itemstack = remapItemToClient(version, locale, itemstack.cloneItemStack());
```

Comment:
wrong position for itemstack remap. Remap is supposed to occur after event, because event adds tags in recent format